File: 백준/Gold/1863. 스카이라인 쉬운거/스카이라인 쉬운거.py
n = int(input())  # 숫자 n 입력 받기
a = []

for _ in range(n):
    x, y = map(int, input().split())  # 튜플의 각 원소 입력 받기
    a.append((x, y))  # 리스트에 튜플 추가

# 리스트 a를 정렬
a.sort()

ans = 0
stk = []

for to in a:
    # 스택에 요소가 있고 스택의 top 요소가 to[1] 보다 크면 pop
    while len(stk) > 0 and stk[-1] > to[1]:
        stk.pop()
    
    # to[1]이 0이면 계속
    if to[1] == 0:
        continue
    
    # 스택이 비어 있거나 스택의 top 요소가 to[1]과 다르면
    if len(stk) == 0 or stk[-1] != to[1]:
        ans += 1
        stk.append(to[1])  # to[1]을 스택에 push

print(ans)

File: 백준/Gold/1863. 스카이라인 쉬운거/스카이라인 쉬운거.cc
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef pair<int, int> pii;
typedef pair<ll, ll> pll;
typedef vector<vector<ll>> vll;
ll gcd(ll a, ll b) { for (; b; a %= b, swap(a, b)); return a; }
int dx[4] = { -1, 1, 0, 0 };
int dy[4] = { 0, 0, -1, 1 };

int main(void)
{
	ios::sync_with_stdio(0);
	cin.tie(0), cout.tie(0);
	ll n; cin >> n;
	vector<pll> a(n);
	for (ll i = 0; i < n; i++)
		cin >> a[i].first >> a[i].second;

	sort(a.begin(), a.end());

	ll ans = 0;
	vector<ll> stk;

	for (ll i = 0; i < n; i++)
	{
		if (a[i].second == 0)
		{
			ans += stk.size();
			stk.clear();
			continue;
		}
		while (stk.size() && stk.back() > a[i].second)
		{
			ans++;
			stk.pop_back();
		}

		if (stk.size() && stk.back() != a[i].second)
			stk.push_back(a[i].second);
		else if (stk.empty())
			stk.push_back(a[i].second);
	}

	ans += stk.size();
	cout << ans;
}